### PR TITLE
Story/17

### PR DESCRIPTION
**Pull request by eliasvagan**
Completes user story 17 for "liking" data entries.

### CODE REVIEW

**Comment by petterroea** [comment]
I don't want to review this, as i wrote the backend. @Letaguator ?

File: backend/Controllers/MetadataController.cs [hunk 1]
```diff
@@ -108,6 +111,64 @@ public async Task<IActionResult> SetExperience([FromBody] SaveExperiencePostReso
...
+				var like = new Like(){LikeUser = user, Metadata = metadata};
+
+				await _likeService.SaveAsync(like);
+			}
+
+			await _unitOfWork.CompleteAsync();
+			return Ok();
+		}
+
+		/// <summary>
```
Comment (Letaguator) [inline backend/Controllers/MetadataController.cs]: This documentation seems off?

Comment (eliasvagan) [reply backend/Controllers/MetadataController.cs]: @petterroea

File: frontend/src/pages/Inspection/MetadataToolbar.jsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  const handleLike = async () => {
+    try {
+      const res = await fetch(`/api/Metadata/${uuid}/like`, {
+        method: 'PUT',
+        headers: {
+          Authorization: `bearer ${token}`,
+        },
+      });
+      const { status } = res;
+      if (status === 200) {
```
Comment (AndreasJJ) [inline frontend/src/pages/Inspection/MetadataToolbar.jsx]: You should have a 401 check too

File: frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx [hunk 3]
```diff
@@ -1,97 +1,167 @@
...
+  const handleLike = async () => {
+    try {
+      const res = await fetch(`/api/Metadata/${uuid}/like`, {
+        method: 'PUT',
+        headers: {
+          Authorization: `bearer ${token}`,
+        },
+      });
+      const { status } = res;
+      if (status === 200) {
```
Comment (AndreasJJ) [inline frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx]: You should have a 401 check too

File: frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx [hunk 4]
```diff
@@ -100,22 +170,22 @@ const SingleMetaDataResult = ({ metadata }) => {
...
       <MetaDataRating>
-        <CommentsIcon href={`URL-TO-METADATA-PAGE-${uuid}`}>
-          <img src={chatIcon} alt="Feedback" />
-          <p>[number] Comments</p>
-        </CommentsIcon>
+        <Favourite onClick={handleLike}>
+          {isLiked ? <StarFullStyled /> : <StarEmptyStyled />}
+          <p>{likes}</p>
+        </Favourite>
+        <CommentsLink to={`viewData/dataset/${uuid}`}>
```
Comment (AndreasJJ) [inline frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx]: This url doesnt work after the splash page got updated

File: frontend/src/pages/Inspection/MetadataToolbar.jsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      <div>
+        {
+          experiencePostGuid
+            ? (
+              <FeedbackLink to={`/articles/${experiencePostGuid}`}>
+                <FeedbackIcon />
+                <p>Experience Article</p>
+              </FeedbackLink>
+            )
+            : <WriteFeedbackLink to={`/articles/new/${uuid}`}>Are you the owner of this data set? Write an experience article here</WriteFeedbackLink>
```
Comment (AndreasJJ) [inline frontend/src/pages/Inspection/MetadataToolbar.jsx]: Skjermbilde 2020-03-05 kl  17 05 59
This isnt centered

Comment (eliasvagan) [reply frontend/src/pages/Inspection/MetadataToolbar.jsx]: It wasn't supposed to be :|

Comment (AndreasJJ) [reply frontend/src/pages/Inspection/MetadataToolbar.jsx]: smh, always center your elements

Comment (eliasvagan) [reply frontend/src/pages/Inspection/MetadataToolbar.jsx]: On it

**Comment by AndreasJJ** [comment]
the reason why it's giving 401 errors by the way is because the way yo get the token is a bit wonky and 'token' is actually the user object. so token.token would be the token.

**Comment by eliasvagan** [comment]
> the reason why it's giving 401 errors by the way is because the way yo get the token is a bit wonky and 'token' is actually the user object. so token.token would be the token.

Replaced in the last patch.

**Comment by Letaguator** [comment]
Looks good to me. Will approve after Andreas comment has been resolved, and @petterroea has viewed my comment.

**Comment by AndreasJJ** [comment]
Elias still needs to update the link that is on the old format.

File: frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx (hunk 4 above)
Comment (petterroea) [reply frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx]: This should have been resolved :thinking:

**Comment by petterroea** [comment]
> I am getting 401 error on like (and that was after deleting sql multiple times (cus docker is shit) and registrating a new user)

Remember that you need to log out and in again after deleting `sql`

File: backend/Controllers/MetadataController.cs (hunk 1 above)
Comment (petterroea) [reply backend/Controllers/MetadataController.cs]: Whoops!

File: frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx (hunk 4 above)
Comment (eliasvagan) [reply frontend/src/pages/MetadataByMunicipality/SingleMetaDataResult.jsx]: Fixed this right now.

**Comment by eliasvagan** [comment]
Ship it

**Pull request merged**